main.go

```diff
@@ -38,30 +38,59 @@ func main() {
 	log.Event(ctx, "loaded configuration", log.INFO, log.Data{"config": cfg})
 
 	// Initialise Health Check?
-	versionInfo, err := health.NewVersionInfo(BuildTime, GitCommit, Version)
+	versionInfo, err := dpHealthCheck.NewVersionInfo(BuildTime, GitCommit, Version)
 	if err != nil {
 		log.Event(ctx, "could not instantiate health check", log.FATAL, log.Error(err))
 		os.Exit(1)
 	}
 
-	hc := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
-	hc.Start(ctx)
+	hc := dpHealthCheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
 
 	// Initialise database
-	var dataStore interfaces.DataStore
-	dataStore = &mongo.Mongo{}
-	err = dataStore.Init(cfg.MongoConfig)
+	//var dataStore interfaces.DataStore
+	mongodb := &mongo.Mongo{}
+	err = mongodb.Init(cfg.MongoConfig)
 	if err != nil {
 		log.Event(ctx, "failed to initialise mongo", log.FATAL, log.Error(err))
 		os.Exit(1)
 	}
 
+	databaseCollectionBuilder := make(map[dpMongoDB.Database][]dpMongoDB.Collection)
+	databaseCollectionBuilder[(dpMongoDB.Database)(mongodb.Database)] = []dpMongoDB.Collection{(dpMongoDB.Collection)(mongodb.Collection)}
+
+	mongoClient := dpMongoDB.NewClientWithCollections(mongodb.Session, databaseCollectionBuilder)
+
+	// Add API checks
+	registerCheckers(ctx, &hc, mongoClient)
+
 	// Initialise server
 	svc := initialiser.Service{}
 	router := mux.NewRouter()
 	svc.Server = initialiser.GetHTTPServer(cfg.BindAddr, router)
 
-	svc.API = api.Setup(ctx, cfg.BindAddr, router, dataStore, &hc)
+	svc.API = api.Setup(ctx, cfg.BindAddr, router, mongodb, &hc)
+
+	hc.Start(ctx)
 
 	svc.Server.ListenAndServe()
+
+	hc.Stop()
+}
+
+// registerCheckers adds the checkers for the provided clients to the health check object
+func registerCheckers(ctx context.Context, hc *dpHealthCheck.HealthCheck, mongoClient *dpMongoDB.Client) {
+	var hasErrors bool
+	mongoHealth := dpMongoDB.CheckMongoClient{
+		Client:      *mongoClient,
+		Healthcheck: mongoClient.Healthcheck,
+	}
+	if err := hc.AddCheck("mongoDB", mongoHealth.Checker); err != nil {
+		hasErrors = true
+		log.Event(ctx, "error adding mongoDB checker", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
+
+	if hasErrors {
+		log.Event(ctx, "error registering checkers for healthcheck", log.ERROR)
```

Comment:
Instead of using os.Exit(1)m return an error here back to main.go

`errors.New("Error(s) registering checkers for healthcheck")`

Then handle the returned error in main.go by exiting the application